extension.ts

```diff
@@ -334,19 +327,27 @@ export async function activate(context: vscode.ExtensionContext) {
           case 'underline':
             options.cursorStyle = vscode.TextEditorCursorStyle.Underline;
             break;
-
           default:
             break;
         }
         editor.options = options;
       });
-      await vscode.commands.executeCommand('setContext', 'vim.active', Globals.active);
+      await vscode.commands.executeCommand('setContext', 'vim.active', false);
       let mh = await getAndUpdateModeHandler();
       mh.setStatusBarText('-- VIM: DISABLED --');
+    } else {
+      await vscode.commands.executeCommand('setContext', 'vim.active', true);
+      compositionState = new CompositionState();
+      modeHandlerToEditorIdentity = {};
+      let mh = await getAndUpdateModeHandler();
+      mh.updateView(mh.vimState, { drawSelection: false, revealRange: false });
     }
-  });
+  }
 
-  vscode.commands.executeCommand('setContext', 'vim.active', Globals.active);
+  registerCommand(context, 'toggleVim', async () => {
+    Configuration.disableExtension = !Configuration.disableExtension;
```

Comment:
I think this change is only in-memory and doesn't actually change the settings to the configuration file. I don't think we ever call the `update` fx in the [api docs](https://code.visualstudio.com/docs/extensionAPI/vscode-api#_a-nameworkspaceconfigurationaspan-classcodeitem-id943workspaceconfigurationspan).

It seems like we should convert `Configuration` class to use getters/setters such that if you set any of the configurations, it'll call the aforementioned update api.

Reply:
For now, for the sake of limiting the scope of these changes, I've only implemented a getter/setter for the `Configuration.disableExtension` property.